frontend/src/components/OdhAppCard.tsx

```diff
@@ -168,7 +168,7 @@ const OdhAppCard: React.FC<OdhAppCardProps> = ({ odhApp }) => {
   return (
     <Card
       component="div"
-      data-id={odhApp.metadata.name}
+      data-testid={odhApp.metadata.name}
```

Comment:
Always assign a static ID to `data-testid`. Additionally once https://github.com/opendatahub-io/odh-dashboard/pull/2579 is fixed, you can also add additional space separated values.
```suggestion
      data-testid={`card ${odhApp.metadata.name}`}
```

Reply:
@christianvogt here  id also dynamically assigned https://github.com/opendatahub-io/odh-dashboard/blob/1b7e2e5a525f26fa20f02d229982967d2cf7131c/frontend/src/components/OdhAppCard.tsx#L172.That is the reason I have assigned same value for data-testid.